### ISSUE TITLE
Wonder about the terms of "input", "output".

### ISSUE THREAD

**Issue by youngmtool** [ProgrammingBlockchain/ProgrammingBlockchain#94]
At this page:
https://github.com/youngmtool/ProgrammingBlockchain/blob/master/bitcoin_transfer/spend_your_coin.md

At this sentence:
/Generate a TxOut for hallOfTheMakers.
TxOut hallOfTheMakersTxOut = new TxOut()

1.
This code is making 2 TxOuts.
But under my understanding, since "in" and "out" in the transaction is TxIn and TxOut, one transaction can have one TxIn and one TxOut.
But each TxIn and TxOut can have multiple inputs and outputs.
See the illustration, TransactionSendReceiveCoins.png, I draw:
https://youngmtool.gitbooks.io/csharpdotnetconceptsandillustrations/blockchain.html

Can I say that the above code is making 2 outputs rather than 2 TxOut?

**Comment by nopara73** [comment]
> But under my understanding, since "in" and "out" in the transaction is TxIn and TxOut, one transaction can have one TxIn and one TxOut.  

That's not correct. I think your confusion comes from how you read json and how Bitcoin Core traditionally named its rpc output. in and out are actually arrays. It would be more correct to call the arrays ins and outs.  

> Can I say that the above code is making 2 outputs rather than 2 TxOut?

TxOut, out, output are synonyms. I'd keep it `TxOut`, since that's what NBitcoin uses.

**Comment by youngmtool** [comment]
Of the sentence "Bitcoin Core traditionally named its rpc output", traditionally?, it means the representation of the transaction in the JSON is now different? Or the representation of the transaction in the JSON, under this sentence: scriptPubKeys of the "normal" TxOuts and the scriptPubKey of the TxOut with the message, at this page : https://github.com/youngmtool/ProgrammingBlockchain/blob/master/bitcoin_transfer/spend_your_coin.md, is still identical with the up to date format?

And again, of above JSON representation of the transaction, so that you mean I can call the key "in" as "array ins"?
But why is "in" pluralized? Under my understanding, one transaction can have only one "in". Is it wrong?

I've never seen the term indicating "in" or "out" in this book, all the terms such as TxOut, output, out indicate each object holding value and ScriptPubKey in the array, so I think "in" and "out" are needed to be mentioned as a term.

**Comment by NicolasDorier** [comment]
I personally prefer TxOut than output when describing code.

**Comment by NicolasDorier** [comment]
the ins pluralized come from core as well.

**Comment by youngmtool** [comment]
@NicolasDorier 
But what do you call for "out" as a key in a transaction because "out" is the term which is already used for indicating TxOut and output?

Bitcoin core, does it mean the upgraded new version of Bitcoin?
Anyway, do you mean that in Bitcoin core multiple "in" in a transaction, like above my wondering not meaning each object in the array which is keyed as "in", can be come?

**Comment by NicolasDorier** [comment]
out == output == TxOut. in == input == TxIn. Bitcoin Core is software implementing Bitcoin protocol.
